File: Signiture-Backend/kotlin/study/src/main/java/function/extension/Exam.java
package function.extension;

public class Exam {

    public Exam(ExamUser examUser) {
        if (ObjectUtils.isNotNull(examUser) && ObjectUtils.isNotNull(examUser.getName())) {
            if (StringUtils.isNotBlank(examUser.getName())) {
                System.out.println("Exam Name: " + examUser.getName());
            }
        }
    }

    public static void main(String[] args) {
        ExamUser nullUser = new ExamUser();
        new Exam(nullUser);

        ExamUser user = new ExamUser("name");
        new Exam(user);
    }

    static class ExamUser {

        private String name;

        public ExamUser() {
        }

        public ExamUser(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }

    class ObjectUtils {

        public static boolean isNotNull(Object value) {
            return value != null;
        }
    }

    class StringUtils {

        public static boolean isNotBlank(String value) {
            return !value.isBlank();
        }
    }
}

File: Signiture-Backend/kotlin/study/src/main/kotlin/function/extension/Exam.kt
package pcy.study.function.extension

fun main() {
    val nullUser = ExamUser()
    exam(nullUser)

    val user = ExamUser("name")
    exam(user)
}

fun exam(examUser: ExamUser?) {

    if(examUser.isNotNull() && examUser?.name.isNotNullOrBlank()) {
        println("Exam Name: ${examUser?.name}")
    }
}

data class ExamUser(var name: String? = null)

fun String?.isNotNullOrBlank(): Boolean {  // 확장 함수 extension function
    return !this.isNullOrBlank()
}

fun Any?.isNotNull(): Boolean {
    return this != null
}